src/include/config-h.in.cmake

```diff
@@ -123,9 +123,22 @@
 /* Accelio conditional compilation */
 #cmakedefine HAVE_XIO
 
+
 /* AsyncMessenger RDMA conditional compilation */
 #cmakedefine HAVE_RDMA
 
+/* define if embedded enabled */
+#cmakedefine WITH_EMBEDDED
```

Comment:
The two flags have different semantics. WITH_EMBEDDED says we are building the embedded library. BUILDING_FOR_EMBEDDED means that the source code (usually a plugin) is currently being compiled for the use of the embedded library, and it should not compile any entry points etc. 

I do think the a single run of make is the best approach here. other wise we would need to run a special build just for embedded. given the small amount of recompile involved I dont believe this is needed. MySql for comparison follows a similar path, compiling twice only the code needed to generate libmysqld (they're cmake based as well).